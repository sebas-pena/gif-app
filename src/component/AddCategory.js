import React, { useState } from 'react'
import PropTypes from 'prop-types'



export const AddCategory = ({ setCategories, categories }) => {

    const [inputValue, setInputValue] = useState('')

    const handleSubmit = (e) => {
        e.preventDefault()
        if (inputValue && !categories.includes(inputValue)) {
            setCategories([inputValue, ...categories])
            setInputValue('')
        }
    }

    const handleInputChange = (e) => setInputValue(e.target.value)


    return (
        <form onSubmit={handleSubmit} >
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                name='category'
                placeholder='Add Category'
            />
        </form>
    )

}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
