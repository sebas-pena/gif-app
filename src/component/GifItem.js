import React from 'react'
import PropTypes from 'prop-types'
export const GifItem = ({ url, title }) => {
    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
}