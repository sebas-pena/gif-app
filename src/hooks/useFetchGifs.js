import { useState, useEffect } from "react"
import { getGifs } from "../helpers/getGifs"



export const useFetchGifs = (category) => {


    const [state, setState] = useState(
        {
            data: [],
            isLoading: true
        }
    );

    useEffect((async () => {
        const data = await getGifs(category)

        setState(
            {
                data,
                isLoading: false
            }
        )

    }), [])

    return state
}