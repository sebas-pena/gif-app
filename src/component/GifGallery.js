import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifItem } from './GifItem'
import PropTypes from 'prop-types'

export const GifGallery = ({ category }) => {

    const { data, isLoading } = useFetchGifs(category)

    return (
        <div className=' animate__animated animate__fadeIn' >
            <h3>{category}</h3>
            <div className='gif-gallery'>
                {isLoading && <p>Loading...</p>}
                {
                    data.map(gif => <GifItem key={gif.id} {...gif} />)
                }
            </div>
            <hr />
        </div>
    )
}


GifGallery.propTypes = {
    category: PropTypes.string.isRequired
}