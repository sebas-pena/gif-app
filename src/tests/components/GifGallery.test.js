import React from 'react'
import { shallow } from 'enzyme'
import { GifGallery } from '../../component/GifGallery'
import { useFetchGifs } from '../../hooks/useFetchGifs'

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas sobre <GifGallery />', () => {


    test('Se debe renderizar correctamente el componente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            isLoading: true
        })

        const wrapper = shallow(<GifGallery category='test' />)
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de mostrar items cuando se cargan las imagenes useFetchGifs', () => {

        const firstGif = {
            id: 9090,
            url: 'http://localhost:80/algo.jpg',
            title: 'titulo del gif'
        }


        const secondGif = {
            id: 9091,
            url: 'http://localhost:80/algo2.jpg',
            title: '2do titulo del gif'
        }

        useFetchGifs.mockReturnValue({
            data: [firstGif, secondGif],
            isLoading: false
        })
        const wrapper = shallow(<GifGallery category='test' />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifItem').length).toBe(2)
        expect(wrapper.find('GifItem').at(0).props()).toEqual(firstGif)

    })

})