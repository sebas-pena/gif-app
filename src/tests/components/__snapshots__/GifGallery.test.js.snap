// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifGallery /> Debe de mostrar items cuando se cargan las imagenes useFetchGifs 1`] = `
<div
  className=" animate__animated animate__fadeIn"
>
  <h3>
    test
  </h3>
  <div
    className="gif-gallery"
  >
    <GifItem
      id={9090}
      key="9090"
      title="titulo del gif"
      url="http://localhost:80/algo.jpg"
    />
    <GifItem
      id={9091}
      key="9091"
      title="2do titulo del gif"
      url="http://localhost:80/algo2.jpg"
    />
  </div>
  <hr />
</div>
`;

exports[`Pruebas sobre <GifGallery /> Se debe renderizar correctamente el componente 1`] = `
<div
  className=" animate__animated animate__fadeIn"
>
  <h3>
    test
  </h3>
  <div
    className="gif-gallery"
  >
    <p>
      Loading...
    </p>
  </div>
  <hr />
</div>
`;
